lib/plugins/aws/deployFunction/index.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const fs = require('fs');
+const SDK = require('../');
+const validate = require('../lib/validate');
+
+// The Package plugin which is used to zip the service
+const Package = require('../../package');
+
+class AwsDeployFunction {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options || {};
+    this.provider = 'aws';
+    this.sdk = new SDK(serverless);
+
+    this.functionName =
+      `${this.serverless.service.service}-${this.options.stage}-${this.options.function}`;
+
+    this.pkg = new Package(this.serverless, this.options);
```

Comment:
can't the package plugin simply be extended to also use a lifecycle of the `deploy function` command? This way we don't need to use it directly and there is no direct dependency here